plugins/language_awk.lua

```diff
@@ -0,0 +1,83 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "Awk script",
+  files = "%.awk$",
+  headers = "^#!.*bin.*awk",
+  comment = "#",
+  patterns = {
+    -- $# is a awk special variable and the '#' shouldn't be interpreted
+    -- as a comment.
+    { pattern = "$[%a_@*#][%w_]*",                type = "keyword2" },
```

Comment:
Escaping this for clarity, even if not needed.
```suggestion
    { pattern = "%$[%a_@*#][%w_]*",               type = "keyword2" },
```